File: src/backend/cadastros/src/model/papel_funcao_model.py
from src import db


class PapelFuncaoModel(db.Model):
    __tablename__ = 'papel_funcao'

    id = db.Column(db.Integer, primary_key=True)
    habilitado = db.Column(db.String(1))
    pode_inserir = db.Column(db.String(1))
    pode_alterar = db.Column(db.String(1))
    pode_excluir = db.Column(db.String(1))
    id_papel = db.Column(db.Integer, db.ForeignKey('papel.id'))
    id_funcao = db.Column(db.Integer, db.ForeignKey('funcao.id'))


    def mapping(self, data):
        self.id = data.get('id')
        self.id_papel = data.get('idPapel')
        self.id_funcao = data.get('idFuncao')
        self.habilitado = data.get('habilitado')
        self.pode_inserir = data.get('podeInserir')
        self.pode_alterar = data.get('podeAlterar')
        self.pode_excluir = data.get('podeExcluir')

    def serialize(self):
        return {
            'id': self.id,
            'idPapel': self.id_papel,
            'idFuncao': self.id_funcao,
            'habilitado': self.habilitado,
            'podeInserir': self.pode_inserir,
            'podeAlterar': self.pode_alterar,
            'podeExcluir': self.pode_excluir,
        }

File: src/frontend/cadastros/lib/app/data/model/papel_funcao_model.dart
import 'dart:convert';
import 'package:pluto_grid/pluto_grid.dart';

import 'package:cadastros/app/data/domain/domain_imports.dart';

class PapelFuncaoModel {
	int? id;
	int? idPapel;
	int? idFuncao;
	String? habilitado;
	String? podeInserir;
	String? podeAlterar;
	String? podeExcluir;

	PapelFuncaoModel({
		this.id,
		this.idPapel,
		this.idFuncao,
		this.habilitado,
		this.podeInserir,
		this.podeAlterar,
		this.podeExcluir,
	});

	static List<String> dbColumns = <String>[
		'id',
		'habilitado',
		'pode_inserir',
		'pode_alterar',
		'pode_excluir',
	];
	
	static List<String> aliasColumns = <String>[
		'Id',
		'Habilitado',
		'Pode Inserir',
		'Pode Alterar',
		'Pode Excluir',
	];

	PapelFuncaoModel.fromJson(Map<String, dynamic> jsonData) {
		id = jsonData['id'];
		idPapel = jsonData['idPapel'];
		idFuncao = jsonData['idFuncao'];
		habilitado = PapelFuncaoDomain.getHabilitado(jsonData['habilitado']);
		podeInserir = PapelFuncaoDomain.getPodeInserir(jsonData['podeInserir']);
		podeAlterar = PapelFuncaoDomain.getPodeAlterar(jsonData['podeAlterar']);
		podeExcluir = PapelFuncaoDomain.getPodeExcluir(jsonData['podeExcluir']);
	}

	Map<String, dynamic> get toJson {
		Map<String, dynamic> jsonData = <String, dynamic>{};

		jsonData['id'] = id != 0 ? id : null;
		jsonData['idPapel'] = idPapel != 0 ? idPapel : null;
		jsonData['idFuncao'] = idFuncao != 0 ? idFuncao : null;
		jsonData['habilitado'] = PapelFuncaoDomain.setHabilitado(habilitado);
		jsonData['podeInserir'] = PapelFuncaoDomain.setPodeInserir(podeInserir);
		jsonData['podeAlterar'] = PapelFuncaoDomain.setPodeAlterar(podeAlterar);
		jsonData['podeExcluir'] = PapelFuncaoDomain.setPodeExcluir(podeExcluir);
	
		return jsonData;
	}
	
	String objectEncodeJson() {
		final jsonData = toJson;
		return json.encode(jsonData);
	}

	plutoRowToObject(PlutoRow plutoRow) {
		id = plutoRow.cells['id']?.value;
		idPapel = plutoRow.cells['idPapel']?.value;
		idFuncao = plutoRow.cells['idFuncao']?.value;
		habilitado = plutoRow.cells['habilitado']?.value != '' ? plutoRow.cells['habilitado']?.value : 'Sim';
		podeInserir = plutoRow.cells['podeInserir']?.value != '' ? plutoRow.cells['podeInserir']?.value : 'Sim';
		podeAlterar = plutoRow.cells['podeAlterar']?.value != '' ? plutoRow.cells['podeAlterar']?.value : 'Sim';
		podeExcluir = plutoRow.cells['podeExcluir']?.value != '' ? plutoRow.cells['podeExcluir']?.value : 'Sim';
	}	

	PapelFuncaoModel clone() {
		return PapelFuncaoModel(
			id: id,
			idPapel: idPapel,
			idFuncao: idFuncao,
			habilitado: habilitado,
			podeInserir: podeInserir,
			podeAlterar: podeAlterar,
			podeExcluir: podeExcluir,
		);			
	}

	
}